src/profile-logic/call-node-info.js

```diff
@@ -222,11 +224,78 @@ export class CallNodeInfoInvertedImpl
   extends CallNodeInfoImpl
   implements CallNodeInfoInverted
 {
+  // This is a Map<SuffixOrderIndex, IndexIntoNonInvertedCallNodeTable>.
+  // It lists the non-inverted call nodes in "suffix order", i.e. ordered by
+  // comparing their call paths from back to front.
+  _suffixOrderedCallNodes: Uint32Array;
+  // This is the inverse of _suffixOrderedCallNodes; i.e. it is a
+  // Map<IndexIntoNonInvertedCallNodeTable, SuffixOrderIndex>.
+  _suffixOrderIndexes: Uint32Array;
+
+  constructor(
+    callNodeTable: CallNodeTable,
+    nonInvertedCallNodeTable: CallNodeTable,
+    stackIndexToCallNodeIndex: Int32Array,
+    stackIndexToNonInvertedCallNodeIndex: Int32Array,
+    suffixOrderedCallNodes: Uint32Array,
+    suffixOrderIndexes: Uint32Array
+  ) {
+    super(
+      callNodeTable,
+      nonInvertedCallNodeTable,
+      stackIndexToCallNodeIndex,
+      stackIndexToNonInvertedCallNodeIndex
+    );
+    this._suffixOrderedCallNodes = suffixOrderedCallNodes;
+    this._suffixOrderIndexes = suffixOrderIndexes;
+  }
+
   isInverted(): boolean {
     return true;
   }
 
   asInverted(): CallNodeInfoInverted | null {
     return this;
   }
+
+  getSuffixOrderedCallNodes(): Uint32Array {
+    return this._suffixOrderedCallNodes;
+  }
+
+  getSuffixOrderIndexes(): Uint32Array {
+    return this._suffixOrderIndexes;
+  }
+
+  getSuffixOrderIndexRangeForCallNode(
+    callNodeIndex: IndexIntoCallNodeTable
+  ): [SuffixOrderIndex, SuffixOrderIndex] {
+    const callPath = this.getCallNodePathFromIndex(callNodeIndex);
+    return bisectEqualRange(
+      this._suffixOrderedCallNodes,
+      (callNodeIndex: IndexIntoCallNodeTable) => {
```

Comment:
> Is it similar to _compareNonInvertedCallNodesInSuffixOrder below, except that we computed the needle's callPath once at the start for a faster access in the iteration, while it would be more costly than necessary for all the iterating callNodeIndexes?

That's right, it's exactly that.

I'm changing this code to just call `_compareNonInvertedCallNodesInSuffixOrder` directly. This code is removed later on in the patch stack anyway, so the premature optimization was simply unnecessary. I'm also renaming the lambda argument to `comparedCallNodeIndex`.

Reply:
Oh that didn't work. `_compareNonInvertedCallNodesInSuffixOrder` takes two non-inverted call nodes. But the argument to `getSuffixOrderIndexRangeForCallNode` is an **inverted** call node. So I'll go back to what I had  before, but with a separate function, more comments, and better variable names.